test/GroupByTest.cpp

```diff
@@ -608,7 +602,8 @@ TEST_F(GroupByOptimizations, checkIfHashMapOptimizationPossible) {
   auto testFailure = [this](const auto& groupByVariables, const auto& aliases,
                             const auto& join, auto& aggregates) {
     auto groupBy = GroupBy{qec, groupByVariables, aliases, join};
-    ASSERT_FALSE(groupBy.checkIfHashMapOptimizationPossible(aggregates));
+    ASSERT_EQ(std::nullopt,
+              groupBy.checkIfHashMapOptimizationPossible(aggregates));
```

Comment:
This would be a nice opportunity to implement
`AD_EXPECT_NULLOPT(singleArgument)` and maybe also `AD_ASSERT_NULLOPT`
(should be simple, we have a `GTestHelpers.h` file which I believe also has its own unit tests.